docs/agents/agent-development.md

```diff
@@ -195,6 +195,24 @@ Service metadata relates to the service/application being monitored:
 
 For official Elastic agents, the agent name should just be the name of the language for which the agent is written, in lower case.
 
+### Cloud Provider Metadata
+
+Cloud provider metadata is collected from local cloud provider metadata services:
+
+- availability_zone
+- account
+  - id
+  - name
+- instance
+  - id
+  - name
+- machine.type
+- project
+  - id
+  - name
+- provider
+- region
+
```

Comment:
Might be helpful to external folks if we linked to the Python implementation here?